lib/utils/createConfig.js

```diff
@@ -14,12 +14,12 @@ function createConfig(config, argv, { port }) {
     options.bonjour = true;
   }
 
-  if (argv.host !== 'localhost' || !options.host) {
+  if (argv.host && (argv.host !== 'localhost' || !options.host)) {
     options.host = argv.host;
   }
 
-  if (argv['allowed-hosts']) {
-    options.allowedHosts = argv['allowed-hosts'].split(',');
+  if (argv.allowedHosts) {
```

Comment:
yargs support all `arguments` in `camelCase` format, so we don't need use hack like `argv['allowed-hosts']`